cmake/linker/ld/target.cmake

```diff
@@ -6,6 +6,14 @@ set(CMAKE_LINKER ${GNULD_LINKER})
 
 set_ifndef(LINKERFLAGPREFIX -Wl)
 
+if((${CMAKE_LINKER} STREQUAL "${CROSS_COMPILE}ld.bfd") OR
+   ${GNULD_LINKER_IS_BFD})
+  # ld.bfd was found so let's explicitly use that for linking, see #32237
+  list(APPEND TOOLCHAIN_LD_FLAGS -fuse-ld=bfd)
```

Comment:
I _think_ moving this here is valid but I'm not entirely sure. IIUC, previously `-fuse-ld=bfd` would only be added when `toolchain_ld_link_elf` is called for a given target (and it's dependencies). By adding this to `TOOLCHAIN_LD_FLAGS` instead (which will then be added via `zephyr_ld_options` when `toolchain_ld_base` is invoked), I think we'll effectively add `-fuse-ld=bfd` everywhere we link. As `toolchain_ld_link_elf` is (IIUC) invoked for each of the link stages (pre0/1, final) it seems like this change should be equivalent--are there use cases where we're both linking other things and truly don't want to use the same linker variant as the "main" `.elfs`?

I don't see anything concerning in https://github.com/zephyrproject-rtos/zephyr/commit/39f06e0195d53ab081bc868c08245558d24112fd where this was first added, but I'm not sure if I'm missing/misunderstanding something.

Reply:
I think this is a fine location.